java-connectors/kafka-connect-gcp-common/src/test/java/io/lenses/streamreactor/connect/gcp/common/auth/GCPServiceBuilderConfigurerTest.java

```diff
@@ -0,0 +1,147 @@
+                .authMode(NoAuthMode.INSTANCE)
+                .host(Optional.empty())
+                .httpRetryConfig(RetryConfig.builder().build())
+                .timeouts(
+                        HttpTimeoutConfig.builder()
+                                .connectionTimeout(Optional.empty())
+                                .socketTimeout(Optional.empty())
+                                .build()
+                );
+
+    }
+
+    @Test
+    void testConfigure_withHostAndProjectIdConfigured() throws IOException {
+        val config = configBuilder
+                .host(Optional.of("example.com"))
+                .projectId(Optional.of("test-project"))
+                .build();
+
+        val builder = createMockBuilder();
+
+        GCPServiceBuilderConfigurer.configure(config, builder);
+
+        assertHostAndProjectIdConfigured(builder, "example.com", "test-project");
+    }
+
+    @Test
+    void testConfigure_withRetrySettingsConfigured() throws IOException {
+        RetryConfig retryConfig = RetryConfig.builder().errorRetryInterval(1000).numberOfRetries(3).build();
+
+        val config = configBuilder.httpRetryConfig(retryConfig)
+                .build();
+
+        val builder = createMockBuilder();
+
+        GCPServiceBuilderConfigurer.configure(config, builder);
+
+        assertRetrySettingsConfigured(builder, 1000, 5000, 3);
+    }
+
+    @Test
+    void testConfigure_withTransportOptionsConfigured() throws IOException {
+        val timeoutConfig = HttpTimeoutConfig.builder()
+                        .socketTimeout(Optional.of(5000L))
+                                .connectionTimeout(Optional.of(3000L))
+                                        .build();
+
+        val config = configBuilder.timeouts(timeoutConfig).build();
+
+        val builder = createMockBuilder();
+
+        GCPServiceBuilderConfigurer.configure(config, builder);
+
+        assertTransportOptionsConfigured(builder, 5000, 3000);
+    }
+
+    @Test
+    void testConfigure_withEmptyConfig() throws IOException {
+        val builder = createMockBuilder();
+
+        val config = configBuilder.build();
+
+        GCPServiceBuilderConfigurer.configure(config, builder);
+
+        // Ensure that no properties are set if configuration is empty
+        verify(builder, never()).setHost(anyString());
+        verify(builder, never()).setProjectId(anyString());
+        verify(builder, times(1)).setRetrySettings(RetrySettings.newBuilder().build());
+        verify(builder, times(1)).setCredentials(NoCredentials.getInstance());
+        verify(builder, never()).setTransportOptions(any());
+    }
+
+    private TestSvcServiceOptionsBuilder createMockBuilder() {
+        return mock(TestSvcServiceOptionsBuilder.class);
+    }
+
+    private void assertHostAndProjectIdConfigured(ServiceOptions.Builder<?, ?, ?> builder, String expectedHost, String expectedProjectId) {
+        verify(builder, times(1)).setHost(expectedHost);
```

Comment:
`times(1)` is by default, no need to add that

Reply:
Tell the guy who wrote [this code](https://github.com/lensesio/stream-reactor/blob/5e77256760421286621ce65b30add823cf4b59fb/java-connectors/kafka-connect-azure-eventhubs/src/test/java/io/lenses/streamreactor/connect/azure/eventhubs/source/AzureConsumerRebalancerListenerTest.java#L57): 😁 

Although, if you don't mind I'd like to keep it as it is, I prefer the explicit specification in this case alongside the `never()`.